### ISSUE TITLE
Inconsistent behaviors in sandbox mode with iframe

### ISSUE THREAD

**Issue by emmkimme** [electron/electron#10470]
* Electron version: 1.6.7
* Operating system: Windows 10

According to the activation of the sandbox, I’m facing inconsistent behaviors with a window containing iframes.

# 1st. issue
### How to reproduce
Have a preload.js (pretty basic)
```
const _ipcRenderer = require('electron').ipcRenderer;
windowLocal.ipcRenderer = _ipcRenderer;
console.log(‘preload file loaded’);
```
A html page with at least one iframe

### Actual behavior
When loading such page, we have the preload file loaded several times :
-	one for the top window
-	one for each iframe
Without the sandbox the preload file is loaded only once whatever the number of iframes.

### Expected behavior
It could make sense to load preload file for each iFrame whatever the mode. We are in position to test in the preload if we are in a iframe (window.frameset) and then adapt our code.

# 2nd issue 
### How to reproduce
In sandbox mode, from the top window and from each iframe I send a specific message to the master using the ipcRenderer.

I build a response channel and the message from the window id and frame id
```
    let message = `I'm here from Window '${windowId}'`;
    let responseChannel = `child-response-${windowId}`;
    if (window.frameElement) {
        message += ` Frame '${window.frameElement.id}'`;
        responseChannel += `-${window.frameElement.id}`;
    }
```

I listen the response on this specific response channel
```
    ipcRenderer.on(responseChannel, (...args) => {
        console.log(…args);
    });
```
I send this specific message to master
`    ipcRenderer.send('child-message', responseChannel, message);
`
The master looks like
```
        ipcMain.on('child-message', (event: Electron.IpcMainEvent, ...args: any[]) => {
            console.log(...args);
            event.sender.send(args[0], 'I got your message !');
        });
```

### Actual behavior
Master receives the expected messages
-	child-response-xxx, I'm here from Window 'xxx’
-	child-response-xxx –yyy, I'm here from Window 'xxx’ Frame ‘yyy’

top window receives the response but Iframes never receive their own responses.

If I listen from the topWindow the response channel of the iframe/s, I receive in the topwindow the message supposed to be sent to the iframe.

### Expected behavior
If we can send ipc message from an iframe we should be able to listen ipc as well.

Unfortunately, my sample is deeply linked to my environment, If needed I can build an independent one.

Regards

**Comment by emmkimme** [comment]
For 2nd issue, does it miss a test on the frame in 
void AtomSandboxedRendererClient::DidCreateScriptContext(
    v8::Handle<v8::Context> context, content::RenderFrame* render_frame) {
++  if (!render_frame->IsMainFrame())
++    return;

**Comment by zcbenz** [comment]
Closed with #11227.
